controllers/complianceeventsapi/server.go

```diff
@@ -380,9 +426,108 @@ func parseQueryArgs(queryArgs url.Values) (*queryOptions, error) {
 		}
 	}
 
+	authorizedClusterNames, includeAllManagedClusters,
+		err := getAuthorizedClusterNames(ctx, db, parsed, userChangedConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	if includeAllManagedClusters {
+		delete(parsed.Filters, "clusters.cluster_id")
+		delete(parsed.Filters, "clusters.name")
+
+		return parsed, nil
+	}
+
+	if len(authorizedClusterNames) == 0 {
+		log.V(1).Info("The service account does not have GET authorization for all")
+
+		return nil, ErrNotAuthorized
+	}
+
+	delete(parsed.Filters, "clusters.cluster_id")
```

Comment:
We actually don't want to do this because even though the authorization has to convert this to a cluster name, the filter should still be on the cluster ID. This is because there could be multiple clusters with the same name but different IDs.